docs/developer/architecture/alerting.rst

```diff
@@ -43,3 +43,263 @@ maintenance operation.
 .. note:: Alertmanager configuration documentation is available here_
 
 .. _here: https://prometheus.io/docs/alerting/configuration/
+
+Design Choices
+--------------
+
+To be able to edit existing rules, add new ones, etc., and in ordrer to keep
+these changes across restorations, upgrades and downgrades, we need to put in
+place some mechanisms to configure Prometheus and Alertmanager and persist
+these configurations.
+
+For the persistence part, we will rely on what has been done for
+:doc:`CSC<configurations>` (Cluster and Services Configurations), and use
+the already defined resources for :term:`Alertmanager` and :term:`Prometheus`.
+
+Custom Prometheus Alert Rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We will use the already existing ``metalk8s-prometheus-config``
+:term:`ConfigMap` to store the :term:`Alertmanager` configuration
+customizations.
+
+Adding custom alert rules will be done editing this :term:`ConfigMap`
+under the ``spec.alertRules`` key in ``config.yaml`` as follows::
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: metalk8s-prometheus-config
+      namespace: metalk8s-monitoring
+    data:
+      config.yaml: |-
+        apiVersion: addons.metalk8s.scality.com
+        kind: PrometheusConfig
+        spec:
+          deployment:
+            replicas: 1
+          alertRules:
+            groups:
+            - name: <AlertGroupName>
+              rules:
+              - alert: <AlertName>
+                annotations:
+                  description: description of what this alert is
+                expr: vector(1)
+                for: 10m
+                labels:
+                  severity: critical
+              - alert: <AnotherAlertName>
+                [...]
+            - name: <AnotherAlertGroupName>
+              [...]
+
+PromQL_ is to be used to define ``expr`` field.
+
+This ``spec.alertRules`` entry will be used to generate through Salt a
+``PrometheusRule`` object named ``metalk8s-custom-rules`` in the
+``metalk8s-monitoring`` namespace, which will be automatically consumed by the
+:term:`Prometheus` Operator to generate the new alert rules.
```

Comment:
I think down the road, the content of  `spec.alertRules` in the YAML above can become too large and unmanageable.

Reply:
It's supposed to be a "slimmed-down" version of the actual set of rules we deploy as part of the `prometheus-operator` "addon" - e.g., we could have a `globalCapacityThreshold` value in there, which controls the percentage from which all capacity-related alerts will be raised.

Granted though, we must be careful not to put everything we deploy in this list.

Reply:
Yes, we will not put each and every rules, most of them are not even relevant to be exposed to the user.